pkg/controllers/nodepool/hash/controller.go

```diff
@@ -36,19 +40,35 @@ var _ operatorcontroller.TypedController[*v1beta1.NodePool] = (*Controller)(nil)
 // Controller is hash controller that constructs a hash based on the fields that are considered for static drift.
 // The hash is placed in the metadata for increased observability and should be found on each object.
 type Controller struct {
-	kubeClient client.Client
+	kubeClient             client.Client
+	apiextensionsInterface clientset.Interface
 }
 
-func NewController(kubeClient client.Client) operatorcontroller.Controller {
+func NewController(kubeClient client.Client, apiextensionsInterface clientset.Interface) operatorcontroller.Controller {
 	return operatorcontroller.Typed[*v1beta1.NodePool](kubeClient, &Controller{
-		kubeClient: kubeClient,
+		kubeClient:             kubeClient,
+		apiextensionsInterface: apiextensionsInterface,
```

Comment:
Can you look into seeing if you can access the CRD through the `kubeClient` (controller-runtime interface). There's nothing in our code that's actually using the `kubernetesInteface` anymore from the operator, so it would be really nice if we just called out to the apiserver every time with the same controller-runtime interface